Tests/Adapter/VarnishCacheTest.php

```diff
@@ -81,4 +81,33 @@ public function testValidToken()
         $cache = new VarnishCache('token', array(), $router, 'ban', $resolver);
         $cache->cacheAction($request);
     }
+
+    public function testRunCommand()
+    {
+        $tmpFile = tempnam(sys_get_temp_dir(), '');
+
+        $cache = new VarnishCache(
+            'token',
+            array(
+                sprintf("echo \"varnishadm -T 10.4.1.62:6082 -S /etc/varnish/secret {{ COMMAND }} '{{ EXPRESSION }}'\" >> %s", $tmpFile),
+                sprintf("echo \"varnishadm -T 10.4.1.66:6082 -S /etc/varnish/secret {{ COMMAND }} '{{ EXPRESSION }}'\" >> %s", $tmpFile),
+            ),
+            $this->getMock('Symfony\Component\Routing\RouterInterface'),
+            'ban'
+        );
+
+        $method = new \ReflectionMethod($cache, 'runCommand');
+        $method->setAccessible(true);
+
+        $this->assertTrue($method->invoke($cache, 'ban', 'req.url ~ \'.*\''));
+
+        $this->assertSame(<<<'CMD'
+varnishadm -T 10.4.1.62:6082 -S /etc/varnish/secret ban 'req.url ~ '.*''
+varnishadm -T 10.4.1.66:6082 -S /etc/varnish/secret ban 'req.url ~ '.*''
+
+CMD
+        , file_get_contents($tmpFile));
+
+        unlink($tmpFile);
```

Comment:
> I can test the file does not exist anymore...even if it's not the initial purpose of the test :) or just add @ but hide a possible issue

I can't think of a reason why it would not exist, but since it's in `/tmp`, it's still possible… but it would be very bad luck b/c it means some other process choose that precise time duration to remove the temporary file.